### PR TITLE
v2.0: Add support for Dall-E 3

### LINKED ISSUE #29: wide images
As of now, only square images can be generated by `dallecli`. How about allowing for [wide images](https://github.com/spdustin/ChatGPT-AutoExpert/blob/main/System%20Prompts.md#dalle) as documented

```
namespace dalle {
// Create images from a text-only prompt.
type text2im = (_: {
// The resolution of the requested image, which can be wide, square, or tall. Use 1024x1024 (square) as the default unless the prompt suggests a wide image, 1792x1024, or a full-body portrait, in which case 1024x1792 (tall) should be used instead. Always include this parameter in the request.
size?: "1792x1024" | "1024x1024" | "1024x1792",
// The user's original image description, potentially modified to abide by the dalle policies. If the user does not suggest a number of captions to create, create four of them. If creating multiple captions, make them as diverse as possible. If the user requested modifications to previous images, the captions should not simply be longer, but rather it should be refactored to integrate the suggestions into each of the captions. Generate no more than 4 images, even if the user requests more.
prompts: string[],
// A list of seeds to use for each prompt. If the user asks to modify a previous image, populate this field with the seed used to generate that image from the image dalle metadata.
seeds?: number[], }) => any;
} // namespace dalle.
```

### PR DESCRIPTION
**Pull request by ctrlcctrlv**
Closes #29.

Many new features.

- Shows images in CLI if you have kitty.
- Supports Dall-E 3.
- Supports wide images.
- Supports HD quality images.
- Supports unlimited № of iterations.
- Fixed many bugs.

image

### CODE REVIEW

**Comment by ctrlcctrlv** [comment]
I also added a way to gracefully cancel iterations by pressing ^C, which doesn't stop the current iteration, but exits before next.

```
📻 Signal SIGINT (2) received,
❌ …will exit after image is generated.
⡈ ⏳ Generating image №14/20…
✅ Image generated!
⏳ Downloading image…
✅ Image downloaded. Size: 5507874 bytes

📝 “Revised” prompt:
    Illustrate a humorous scenario in a tailor
shop. On the left is an exceedingly rotund green
raccoon, standing on a raised platform. His
stomach is massive, appearing like an enormous
balloon protruding outside his blue jeans. With
thick arms and legs, he looks very overweight. His
face expresses surprise, with wide eyes and
slightly parted mouth, for the tailor's measuring
tape is a tad too snug around him. He stands about
130 cm tall and weighs a whopping 75 kg, with a
waistline of 140 cm and a body mass index of 49.
On the right is a Japanese White Tiger dressed as
a tailor, sporting a white formal shirt with a
waistcoat and grey trousers. The tiger's tail is
decorated with long black stripes, and its face
bears intricate red patterns, resembling a mask.
The raccoon has a comic 'pop' sound bubble on his
face, as if the tightness of the measuring tape
caused a sound, amplifying his surprise. The 'hmm'
sound insinuates the tape is too short to measure
his plump waist, alluding to the difficulties in
accommodating such a chubby client. To accentuate
the comedic effect, the tape is drawn tightly.

⠀ ⏳ Showing image №14/20…













✅ Done! Image saved to /home/fred/Workspace/openai/images/output_88.png
🛑 Will not generate remaining 6 images due to interruption.
⏱️ Took 0 hours, 7 minutes and 34 seconds to generate 14 images,
✅ of which 7 succeeded and 7 failed.
🤔 Success rate: 50.0%
📝 Note: 6 images were not generated due to Unix signaling.
(venv) [fred@デブ.狸.agency ~/Workspace/openai]$

File: dallecli/cli.py [hunk 1]
```diff
@@ -1,17 +1,50 @@
...
+    import click
+    from PIL import Image, ImageEnhance, ImageFilter, ImageOps, ExifTags
+    from PIL.Image import Exif
+    import openai
+    from openai import OpenAI, AuthenticationError
+    from openai.types import ImagesResponse
+    import requests
+    from rich.console import Console
+except ImportError as e:
+    print(f"❌ Error importing module: {e.path}")
```
Comment (real-or-random) [inline dallecli/cli.py:33]:
```suggestion
    print(f"❌ Error importing module: {e}")
```

File: dallecli/cli.py [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
+import webbrowser
+
+try:
+    import click
+    from PIL import Image, ImageEnhance, ImageFilter, ImageOps, ExifTags
+    from PIL.Image import Exif
+    import openai
+    from openai import OpenAI, AuthenticationError
+    from openai.types import ImagesResponse
+    import requests
```
Comment (real-or-random) [inline dallecli/cli.py:30]: `requests` should be listed in `setup.py` as a dependency

**Comment by ctrlcctrlv** [comment]
thanks @real-or-random for review but i think this project is abandoned

**Comment by raiyanyahya** [comment]
my apologies the project is not abandoned. I will merge the PR and release a new version tomorrow. Thank you so much for your patience.

**Comment by raiyanyahya** [comment]
Hi just saw the changes..thank you for investing time into this. I do not want to use the "pickle" module and hence could not merge the PR. I have however released an updated version of the cli with support for dalle3 and updated the documentation as well. Thank you again and my apologies for the delay.

**Pull request closed without merging**